DokuwikiXMLExport.php

```diff
@@ -0,0 +1,194 @@
+<?php
+
+/**
+ *  This is the Dokuwiki export made by D. Brader for FINDOLOGIC.
+ *  If any bugs occur, please contact the support team (support@findologic.com).
+ */
+if(!defined('DOKU_INC')) {
+    define('DOKU_INC',realpath(dirname(__FILE__).'/../../../').'/');
+}
+
+require_once (DOKU_INC.'inc/init.php');
+require (__DIR__ . '/vendor/autoload.php');
+
+use FINDOLOGIC\Export\Exporter;
+use FINDOLOGIC\Export\Data\Name;
+use FINDOLOGIC\Export\Data\Summary;
+use FINDOLOGIC\Export\Data\Description;
+use FINDOLOGIC\Export\Data\Price;
+use FINDOLOGIC\Export\Data\Url;
+use FINDOLOGIC\Export\Data\Ordernumber;
+use FINDOLOGIC\Export\Data\DateAdded;
+use FINDOLOGIC\Export\Data\Attribute;
+
+class DokuwikiXMLExport
+{
+    /**
+     * Generate the entire XML Export based on the Dokuwiki metadata.
+     *
+     * @param $start integer Determines the first item (offset) to be exported.
+     * @param $count integer Determines the interval size / number of items to be exported.
+     * @return string Returns the XML.
+     */
+    public function generateXMLExport($start, $count)
+    {
+        global $conf;
+
+        $Indexer = new Doku_Indexer();
+        $pagesAndDeletedPages = $Indexer->getPages();
+
+        $excludedPages = $this->splitConfigToArray($conf['plugin']['findologicxmlexport']['excludePages']);
+        $pages = null;
+        foreach ($pagesAndDeletedPages as $page){
+            if (p_get_metadata($page)['description']){ // Only get pages with content
+                if (!in_array($page, $excludedPages)){ // Exclude pages from config
+                    $pages[] = $page;
+                }
+            }
+        }
+        $exporter = Exporter::create(Exporter::TYPE_XML, $count);
+
+        $total = count($pages);
+
+        // If count is higher then total, set count=total to prevent an exception.
+        if ($count > $total) {
```

Comment:
You could just write `$count = max($count, $total);`

Reply:
TODO

- This solution is good. Thanks for that @georgms :+1:

Reply:
@georgms I changed your suggestion `$count = max($count, $total);` to `$count = min($count, $total);`. Because this always gives me the lowest values of one of both.